yocto-update-strategies.md

```diff
@@ -0,0 +1,651 @@
+  - any customizations goes into `meta-custom_layer`: we have full control on
+    what and why was changed through `.bbappends`
+* For `external` we should pick Yocto release; specific commit hash
+  - task for that should be short, e.g. when using NXP BSP as base, we should
+    follow hashes used in given BSP release
+
+???
+
+Never modify the POKY layer. Never. Ever. - even expand it on other layers
+
+---
+# Performing update - custom layer update
+
+* Most of the effort of performing updates should be implemented in custom
+  layers
+* We can list some of the basic tasks carried out when updating Yocto, which
+  should be done in custom layers
+  - updating `LAYERSERIES_COMPAT` variable
+  - going through the bbappends files and verifying that they all still apply
+    (whether they depended on a revised version of a recipe)
+  - verification of custom patches for used packages, perhaps some of them were
+    added in the upstream
+  - syntax and variables verification
+  - path verification for the keywords `require` and `include`
+* Check the migration guides in case we encounter unexpected behavior
+  - there is one per every release
+  - we may found there usefull scripts, like one for override syntax change
+
+---
+# Build troubleshooting
+
+* We can split up every operating system into couple pieces
+.center[<img src="img/boot-process.svg" width="400px">]
+* Updating whole system, means that every part needs to work correctly (meaning,
+  as expected, presented in requirements)
+  - bootloader - configuration, environment variables, boot script,
+    functionality, drivers, boot-time
+  - kernel/device-tree - hardware description, configuration, drivers, boot-time
+  - custom/default apps - configuration, compilation, installation
+  - functionalities - testing, feedback, improvements
+* Other than that, we may also need to update our tests suites and CI/CD
+  pipelines
+
+---
+# Build troubleshooting
+
+* In the following slides, we want to present some of the problems we have had
+  to work on in the recent past
+  - most of them consist of updating to Yocto Kirkstone release
+  - our experience allows us to describe a wide range of problems that any Yocto
+    developer may encounter when updating a custom system
+* We will discuss the following problems
+  - changes in bootloaders and kernels on i.MX based platforms
+  - proper configuration of default apps provided to our systems
+  - modifications to the source code of custom applications when updating
+    systems that use Qt
+  - fixing CI/CD pipelines after switching to new Yocto release
+  - going through test suites to preserve functionalities checking
+
+---
+# Build troubleshooting - bootloader
+
+* Problems with updating bootloader may consist of
+  - correcting bootscript
+  - properly rebasing custom drivers and defconfigs
+  - Yocto recipe compilation, but should not be a big issue as long as we are
```

Comment:
conisder dropping `but should not be a big issue as long as we are
    able to compile it manually`

Reply:
moved to notes https://github.com/3mdeb/conferences/pull/7/commits/0eab34f24aebae4de8e465a7c63e0c3d8bc543f4#diff-a1666090e402e433c955cc46797dc399867480b4d815574efd9f5d4181e24ac3R441